transport/grpc/config.go

```diff
@@ -182,6 +197,45 @@ func (c OutboundTLSConfig) dialOptions() []DialOption {
 	return []DialOption{option}
 }
 
+// OutboundKeepaliveConfig configures gRPC keepalive for a gRPC outbound.
+type OutboundKeepaliveConfig struct {
+	Enabled             bool   `config:"enabled"`
+	Time                string `config:"time"`
+	Timeout             string `config:"timeout"`
+	PermitWithoutStream bool   `config:"permit-without-stream"`
+}
+
+func (c OutboundKeepaliveConfig) dialOptions() ([]DialOption, error) {
+	if !c.Enabled {
+		return nil, nil
+	}
+
+	var err error
+
+	keepaliveTime := time.Second * 30
```

Comment:
Why did you choose 30 seconds as default? Please add a comment

Reply:
Added comments for time and timeout
```
gRPC keepalive expects time to be minimum 10s.
read more: https://pkg.go.dev/google.golang.org/grpc/keepalive#ClientParameters
```

Reply:
Should we put the default as gRPC?

Reply:
Yes, I did that along with comments change. 
* Time - 10s
* Timeout - 20s